### ISSUE TITLE
FileSystemWatcher to a directory symlink that points to a file doesn't raise exception on windows 8.1 or lower

### ISSUE THREAD

**Issue by jozkee** [dotnet/runtime#53010]
### Description
On cmd:
```cmd
# create file
echo text > file.txt 

# create directory symlink to file
mklink /d linktofile file.txt
```

On .NET:
```cs
using System;
using System.IO;

namespace fsw
{
    class Program
    {
        static void Main(string[] args)
        {
            using var watcher = new FileSystemWatcher("linktofile");
            watcher.EnableRaisingEvents = true; // This should've thrown.

            Console.WriteLine("Watching...");
            Console.ReadKey();
        }
    }
}
```

### Configuration

.NET 5 and .NET 6 preview
OS: Windows 8.1 or 7. 
Windows 10 throws which is what I expected since a directory symlink to a file is a bad symlink.

### Regression?

No

### Other information

The only problem I see is consistency.

cc @carlossanlop

**Comment by carlossanlop** [comment]
Were you able to determine the root cause?

**Comment by Jozkee** [comment]
[`CreateFileW`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew) with `FILE_FLAG_BACKUP_SEMANTICS` attribute throws if the path is to a self-referencing directory symlinkin windows 10. In 7 and 8.1, it does not.

**Comment by carlossanlop** [comment]
Thanks. So if those Windows versions do not throw, are you suggesting we detect the self-referencing dir symlink ourselves?

**Comment by Jozkee** [comment]
Yes, it could be an option. We most probably want to only do that for those versions and avoid the manual validation (potentially involves a sys-call) in win10, which currently throws as expected. 

Another option is to find an alternative API that throws in such case and that also returns a handle for a directory, not sure if exists.
